src/frontend/src/expr/table_function.rs

```diff
@@ -226,6 +229,132 @@ impl TableFunction {
         })
     }
 
+    pub fn new_postgres_query(args: Vec<ExprImpl>) -> RwResult<Self> {
+        let evaled_args = {
+            let mut evaled_args: Vec<String> = Vec::with_capacity(6);
+            if args.len() != 6 {
+                return Err(BindError("table_scan function only accepts 6 arguments: postgres_query(hostname varchar, port varchar, username varchar, password varchar, database_name varchar, postgres_query varchar)".to_string()).into());
+            }
+            for arg in &args {
+                if arg.return_type() != DataType::Varchar {
```

Comment:
Also fyi in order to to check argument types prefer `cast_implicit(expected_type)` over `return_type() == expected_type`. There is no difference when `expected_type` is varchar because it is the default/fallback, but using `cast_implicit` allows passing `null` as integer and `'2024-10-10'` as date.